src/core/Elsa.Abstractions/Models/WorkflowPersistenceBehavior.cs

```diff
@@ -6,15 +6,21 @@ namespace Elsa.Models
     [JsonConverter(typeof(StringEnumConverter))]
     public enum WorkflowPersistenceBehavior
     {
+        // TODO: Consider a new service to "get the default persistence behaviour".
+        // Currently it is set to WorkflowBurst implicitly because it is the default
+        // for the enum type.  This would be more clear if explicit.
+        // This includes altering all locations which need a default persistence
+        // behaviour though.
+
         /// <summary>
-        /// Workflow instances are persisted only when being suspended. 
+        /// Workflow instances are persisted after the workflow completed a burst of execution.
         /// </summary>
-        Suspended,
-        
+        WorkflowBurst = 0,
+
         /// <summary>
-        /// Workflow instances are persisted after the workflow completed a burst of execution.
+        /// Workflow instances are persisted only when being suspended. 
         /// </summary>
-        WorkflowBurst,
+        Suspended,
```

Comment:
Indeed, let's not change the order like this, but instead explicitly set the default behavior when constructing a workflow blueprint.

When building workflows using the builder API, this means setting the default behavior on `WorkflowBuilder`.
When building workflows using the designer, this means updating the designer to an empty workflow instance with the default behavior.

The reason I prefer not to change the order of the enum values is because they go from "persists lazily" --> "persist eagerly"

PS: You made a typo in your comment ;)

Reply:
I'm glad you asked about this, because it's the exact same as I was thinking.

> When building workflows using the designer, this means updating the designer to an empty workflow instance with the default behavior.

Do you have any hints as to where the code change needs to be made in order to the bit where you suggest _"updating the designer to an empty workflow instance"_?  I've not even looked at the designer/dashboard/UI stuff at all yet and I know it's currently in flux so I've not been rushing to look into it yet either.  So any pointer to the right code welcome thanks :)

Otherwise, totally, I can make the change in the builder class.  Are there any other places you can think of which might initialise a persistence behaviour?

PS:  It actually took me a few re-reads to remember that US English doesn't include a U in the word "behaviour" 😆   Goes all the way back to things I was taught in childhood to remember the spelling:  "It's not **my** behaviour, it's **our** behaviour".  Thankfully I'm deleting that comment and reverting that part of the change anyway.

Reply:
Sure, it's right here on [this line](https://github.com/elsa-workflows/elsa-core/blob/feature/elsa-2.0/src/designer/elsa-workflows-studio/src/components/designers/tree/elsa-designer-tree/elsa-designer-tree.tsx#L18).

But feel free to skip it, I can just as easily add & test it.

PS: Hahaha sorry, I just couldn't help myself 😄

Reply:
Done